File: 0020有效的括号.py
s="[]"

# 栈
def isValid(s: str) -> bool:
    bracket_map={"}":"{","]":"[",")":"("}
    stack=[]
    
    for i in s:
        if(i not in bracket_map):
            stack.append(i)
            
        else:
            if(not stack):
                return False

            top=stack.pop()
            if(bracket_map[i]!=top):
                return False

    return not stack
    
print(isValid(s))

File: 0020有效的括号.cpp
#include <iostream>
#include <vector>
#include<stack>
#include<map>
using namespace std;


// 栈
bool isValid(string s)
{
    stack<char> st;
    map<char, char> brackets_map = {
        {']','['},
        {')','('},
        {'}','{'},
    };

    for (int i = 0; s[i] != '\0'; ++i)
    {
        if(s[i]=='('||s[i]=='['||s[i]=='{')
            st.push(s[i]);
        else
        {
            if(not st.size())
                return false;
            char top_val = st.top();
            st.pop();
            if(top_val!=brackets_map[s[i]])
                return false;
        }
    }

    return !st.size();
}

int main()
{
    string s = "";
    cout << isValid(s);
}